tsconfig.json

```diff
@@ -0,0 +1,112 @@
+{
```

Comment:
I added a jsconfig.json file to the root of the repo to enable some VS Code in-IDE typechecking.

That should be redundant with this tsconfig file.  Could you remove that in this PR?